resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosFineGrainedSchedulerBackend.scala

```diff
@@ -446,6 +446,8 @@ private[spark] class MesosFineGrainedSchedulerBackend(
 
   // TODO: query Mesos for number of cores
   override def defaultParallelism(): Int = sc.conf.getInt("spark.default.parallelism", 8)
+  override def numCores(): Int = defaultParallelism
+  override def numExecutors(): Int = 1
```

Comment:
Eh, is it simply because it's not implemented yet in Mesos? I am not used to it but I thought it's better be mentioned somewhere.

Reply:
I found this ticket for the TODO: https://issues.apache.org/jira/browse/SPARK-9775 . Should I add a comment to it or create a separate ticket?

Reply:
I added comments to the ticket above.

Reply:
The mode (and the file) is going to be removed in Spark 3.0: https://github.com/apache/spark/pull/18784